File: app/src/main/java/com/example/ryadar/tablayoutmaterialdesign/MainActivity.java
package com.example.ryadar.tablayoutmaterialdesign;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.support.design.widget.TabLayout;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

       ViewPager  viewPager= findViewById(R.id.viewpager);

       TabLayout tabLayout= findViewById(R.id.tabs);

       FragmentPageAdapter adapter = new FragmentPageAdapter(getSupportFragmentManager());

        // Set the adapter onto the view pager
        viewPager.setAdapter(adapter);

        // Give the TabLayout the ViewPager
        tabLayout.setupWithViewPager(viewPager);

    }
}

File: app/src/main/java/com/example/ryadar/tablayoutmaterialdesign/kt/MainActivity.kt
package com.example.ryadar.tablayoutmaterialdesign.kt
 import android.os.Bundle
import android.support.design.widget.TabLayout
 import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import com.example.ryadar.tablayoutmaterialdesign.R



class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?){
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
       var   viewPager : ViewPager= findViewById<ViewPager>(R.id.viewpager)

        var   tabLayout : TabLayout= findViewById<TabLayout>(R.id.tabs)

        var adapter:MyFragmentPageAdapter =MyFragmentPageAdapter(getSupportFragmentManager())
        viewPager.adapter= adapter

        tabLayout.setupWithViewPager(viewPager)

    }

}